e2e-tests/tests/remix.test.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable jest/expect-expect */
+import { Integration } from '../../lib/Constants';
+import {
+  checkEnvBuildPlugin,
+  cleanupGit,
+  KEYS,
+  revertLocalChanges,
+} from '../utils';
+import { startWizardInstance } from '../utils';
+import {
+  checkFileContents,
+  checkFileExists,
+  checkIfBuilds,
+  checkIfRunsOnDevMode,
+  checkIfRunsOnProdMode,
+  checkPackageJson,
+  TEST_ARGS,
+} from '../utils';
+import * as path from 'path';
+
+describe('Remix', () => {
+  const integration = Integration.remix;
+  const projectDir = path.resolve(
+    __dirname,
+    '../test-applications/remix-test-app',
+  );
+
+  beforeAll(async () => {
+    const wizardInstance = startWizardInstance(integration, projectDir);
+    const packageManagerPrompted = await wizardInstance.waitForOutput(
+      'Please select your package manager.',
+      10_000,
+      true,
+    );
+
+    if (packageManagerPrompted) {
+      // Selecting `yarn` as the package manager
+      wizardInstance.sendStdin(KEYS.DOWN);
+      wizardInstance.sendStdin(KEYS.ENTER);
+    }
+
+    const tracingOptionPrompted = await wizardInstance.waitForOutput(
+      'Do you want to enable Tracing',
+      240_000,
+    );
+
+    if (tracingOptionPrompted) {
+      wizardInstance.sendStdin(KEYS.ENTER);
+    }
+
+    const replayOptionPrompted = await wizardInstance.waitForOutput(
+      'Do you want to enable Sentry Session Replay',
+      240_000,
+    );
+
+    if (replayOptionPrompted) {
+      wizardInstance.sendStdin(KEYS.ENTER);
+    }
+
+    const examplePagePrompted = await wizardInstance.waitForOutput(
```

Comment:
I don't fully understand why we need the `optional` configuration... Would you mind explaining it to me? I feel like I'm missing something. 😄

Reply:
It's primarily for local development where the test flows can be fragile, for example when the package manager is detected somehow and it's not prompted, or the leftovers from the example page exist and it's not prompted.

I think we can improve the utils in follow-up PRs to make them more debuggable, but this makes writing / debugging tests slightly easier in the meantime.

Reply:
Seems good! 👍